nexus/src/app/certificate.rs

```diff
@@ -0,0 +1,230 @@
+            params,
+        );
+
+        // TODO: Saga?
+        info!(self.log, "Creating certificate");
+        let cert = self
+            .db_datastore
+            .certificate_create(opctx, new_certificate)
+            .await?;
+        // TODO: Refresh other nexus servers?
+        self.refresh_tls_config(&opctx).await?;
+        info!(self.log, "TLS refreshed successfully");
+        Ok(cert)
+    }
+
+    pub async fn certificates_list(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Name>,
+    ) -> ListResultVec<db::model::Certificate> {
+        self.db_datastore
+            .certificate_list_for(
+                opctx,
+                db::model::ServiceKind::Nexus,
+                pagparams,
+            )
+            .await
+    }
+
+    pub async fn certificate_delete(
+        self: &Arc<Self>,
+        opctx: &OpContext,
+        certificate_lookup: lookup::Certificate<'_>,
+    ) -> DeleteResult {
+        let (.., authz_cert, _db_cert) = certificate_lookup.fetch().await?;
+        self.db_datastore.certificate_delete(opctx, &authz_cert).await?;
+        // TODO: Refresh other nexus servers?
+        self.refresh_tls_config(&opctx).await?;
+        Ok(())
+    }
+
+    // Helper functions used by Nexus when managing its own server.
+
+    /// Returns the dropshot TLS configuration to run the Nexus external server.
+    pub async fn get_nexus_tls_config(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<Option<dropshot::ConfigTls>, Error> {
+        // Lookup x509 certificates which might be stored in CRDB, specifically
+        // for launching the Nexus service.
+        //
+        // We only grab one certificate (see: the "limit" argument) because
+        // we're currently fine just using whatever certificate happens to be
+        // available (as long as it's for Nexus).
+        let certs = self
+            .datastore()
+            .certificate_list_for(
+                &opctx,
+                db::model::ServiceKind::Nexus,
+                &DataPageParams {
+                    marker: None,
+                    direction: dropshot::PaginationOrder::Ascending,
+                    limit: std::num::NonZeroU32::new(1).unwrap(),
+                },
+            )
+            .await
+            .map_err(|e| {
+                Error::internal_error(&format!(
+                    "Failed to list certificates: {e}"
+                ))
+            })?;
+
+        let certificate = if let Some(certificate) = certs.get(0) {
+            certificate
+        } else {
+            return Ok(None);
+        };
+
+        Ok(Some(dropshot::ConfigTls::AsBytes {
+            certs: certificate.cert.clone(),
+            key: certificate.key.clone(),
+        }))
+    }
+
+    /// Refreshes the TLS configuration for the currently-running Nexus external
+    /// server. This involves either:
+    /// - Creating a new HTTPS server if once does not exist
```

Comment:
```suggestion
    /// - Creating a new HTTPS server if one does not exist
```

Reply:
Fixed in 1269e18634b24b0835e54efc9518c7db4cf51c45